### ISSUE TITLE
lib/web/mage/dropdowns.js fails when autoclose is set to true

### ISSUE THREAD

**Issue by brian-labelle** [magento/magento2#15469]
Magento's lib/web/mage/dropdowns.js has a bug that prevents a developer from specifying `autoclose: false` as an option for the dropdown function.

### Preconditions

1. Magento 2.2 or greater
2. Any version of PHP, MySQL
3. Any Browser

### Steps to reproduce

1. Call function 'dropdown' on a jquery element, specifying 'autoclose: false' as an option
```
$(this.selector).dropdown({
    autoclose: false
});
```
2. Open the dropdown
3. Click anywhere inside the dropdown content 

### Expected result

1. The dropdown should remain open

### Actual result

1. Console error indicating we cannot call function hasClass on undefined
   1. `el` is undefined

**Comment by brian-labelle** [comment]
See lines 129-143 of lib/web/mage/dropdowns.js
```
elem.on('click.toggleDropdown', function () {
	var el;

	if (options.autoclose === true) {
		el = actionElem;

		actionElem = $();
		$(document).trigger('click.hideDropdown');
		actionElem = el;
	}

	self[el.hasClass(options.activeClass) ? 'closeDropdown' : 'openDropdown'](elem);

	return false;
});
```

`el` is only initialized as `actionElem`, which should be `$(this)`,  (the element we are calling dropdown on), when `options.autoclose` is `true`

As a result, if `options.autoclose` is `false`, `el` is `undefined`, so we cannot call function `hasClass` on it

**Comment by magento-engcom-team** [comment]
Hi @brian-labelle. Thank you for your report. 
The issue has been fixed in magento/magento2#15499 by @brian-labelle in 2.2-develop branch
Related commit(s):
 - [31e3b63b2fa812f845a4461ce44d1fbdea430956](https://github.com/magento/magento2/commit/31e3b63b2fa812f845a4461ce44d1fbdea430956)

The fix will be available with the upcoming 2.2.5 release.

**Comment by magento-engcom-team** [comment]
Hi @brian-labelle. Thank you for your report. 
The issue has been fixed in magento/magento2#15607 by @dmytro-ch in 2.3-develop branch
Related commit(s):
 - [1e2019ccb7f39c4158f77258cec6d52f32011d70](https://github.com/magento/magento2/commit/1e2019ccb7f39c4158f77258cec6d52f32011d70)

The fix will be available with the upcoming 2.3.0 release.
